core/src/test/java/org/apache/iceberg/hadoop/TestHadoopCatalog.java

```diff
@@ -221,6 +221,22 @@ public void testBasicCatalog() throws Exception {
     Assertions.assertThat(fs.isDirectory(new Path(metaLocation))).isFalse();
   }
 
+  @Test
+  public void testHadoopFileIOProperties() {
+    TableIdentifier tableIdent = TableIdentifier.of("db", "ns1", "ns2", "tbl");
+    ImmutableMap<String, String> catalogProps =
+        ImmutableMap.of(
+            "warehouse", "/hive/testwarehouse",
+            "io.manifest.cache-enabled", "true");
+
+    HadoopCatalog catalog = new HadoopCatalog();
+    catalog.initialize("hadoop", catalogProps);
+    FileIO fileIO = catalog.newTableOps(tableIdent).io();
+
+    Assertions.assertThat(fileIO.properties().get("warehouse")).isEqualTo("/hive/testwarehouse");
```

Comment:
```suggestion
    Assertions.assertThat(fileIO.properties()).containsEntry("warehouse", "/hive/testwarehouse");
```

Reply:
this is generally better, because if the assertion ever fails, it will show the content of properties()